src/markdown_generators.js

```diff
@@ -227,19 +227,41 @@ const getResourceType = mimeType => {
   }
 }
 
-const generateResourceMarkdownForFile = file => {
+const generateResourceMarkdownForFile = (file, courseData) => {
   /**
   Generate the front matter metadata for a PDF file
   */
+  const uid = helpers.addDashesToUid(file["uid"])
   const frontMatter = {
     title:        file["title"],
     description:  file["description"],
-    uid:          helpers.addDashesToUid(file["uid"]),
+    uid:          uid,
     resourcetype: getResourceType(file["file_type"]),
     file_type:    file["file_type"],
     file:         helpers.stripS3(file["file_location"])
   }
 
+  if (frontMatter.resourcetype === RESOURCE_TYPE_IMAGE) {
+    const courseImageUid = helpers.getUidFromFilePath(courseData["image_src"])
+    const courseImageThumbnailUid = helpers.getUidFromFilePath(
+      courseData["thumbnail_image_src"]
+    )
+    const isCourseImage =
+      courseImageThumbnailUid === uid || courseImageUid === uid
+    const alt = isCourseImage
+      ? courseData["image_alternate_text"]
+      : file["alt_text"]
+    const caption = isCourseImage
+      ? courseData["image_caption_text"]
+      : file["caption"]
+
+    frontMatter["image_metadata"] = {
+      ["image-alt"]: alt || "",
+      caption:       caption || "",
```

Comment:
Can we wrap this in `html2markdown(fixLinks(`?  I tested converting all courses and a number of the captions not only have HTML in them but also have `resolveuid` links in them.  Raw HTML in these fields will come through as raw text if this is rendered in Hugo.

Reply:
Sure, I'll also do this for the legacy JSON